drivers/soundwire/generic_bandwidth_allocation.c

```diff
@@ -506,6 +508,12 @@ static int sdw_compute_bus_params(struct sdw_bus *bus)
 		}
 
 		if (!use_multi_lane) {
+			if (i < clk_values) {
+				dev_dbg(bus->dev, "%s: could not find frame configuration for bus dr_freq %d try higher clock rate\n",
+					__func__, curr_dr_freq);
+				i++;
+				goto select_clk;
+			}
```

Comment:
I am not sure why you would only increase the clock when there is a single lane. The clock should be increased even in the multi-lane case when there's no bandwidth left?

I must be missing something.

Reply:
That is about the priority of using multi-lane or higher clock rate. At this point, we tried multi-lane but can't find an available lane and it will return -EINVAL if it reaches the highest clock rate. But yeah, I didn't consider about using multi-lane with higher clock rate. I assume that we always have enough bandwidth if we can find an extra lane. Let me redo this.

Reply:
yes, lanes should be used before cranking the speed up, but an error should only be thrown when multilane+higher speed combined doesn't support the required bandwidth.